images/base/Dockerfile

```diff
@@ -62,9 +62,9 @@ RUN clean-install \
 # - add the fingerprint
 # - add the repository
 # - update apt, install docker, cleanup
-# NOTE: 18.06 is officially supported by Kubernetes currently, so we pin to that.
+# NOTE: 18.09 is officially supported by Kubernetes currently, so we pin to that.
 # https://kubernetes.io/docs/tasks/tools/install-kubeadm/
-ARG DOCKER_VERSION="18.06.*"
+ARG DOCKER_VERSION="5:18.09.*"
```

Comment:
is this `5:` intentional? what does it do?

Reply:
the version is `Version: 5:18.09.4~3-0~ubuntu-bionic` for some reason in the packages... 18.06 did not have a prefix

```
root@kind-control-plane:/# apt-cache madison docker-ce
 docker-ce | 5:18.09.4~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.3~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.2~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.1~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 5:18.09.0~3-0~ubuntu-bionic | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.3~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.2~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.1~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.06.0~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
 docker-ce | 18.03.1~ce~3-0~ubuntu | https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
```

Reply:
https://readme.phys.ethz.ch/documentation/debian_version_numbers/
> Package with epoch, e.g. if upstream changed from date based to major.minor versioning
Old upstream version: 2013.06.06
New upstream version: 1.0
Problem: 1.0 < 2013.06.06
Solution: Add a (debian-specific) epoch:
Old package version (example): 2013.06.06-4 (upstream version + debian revision + implicit epoch of "0", i.e. 0:2013.06.06 is equivalent)
New package version: 1:1.0-1 (epoch of 1 + upstream version + debian revision)

o_O

Reply:
interesting @neolit123 @dims, thanks.